src/apify/scrapy/scheduler.py

```diff
@@ -52,8 +54,15 @@ def open(self, spider: Spider) -> None:  # this has to be named "open"
         self.spider = spider
 
         async def open_queue() -> RequestQueue:
-            custom_loop_apify_client = ApifyStorageClient(configuration=Configuration.get_global_configuration())
-            return await RequestQueue.open(storage_client=custom_loop_apify_client)
+            config = Configuration.get_global_configuration()
+
+            # Use the ApifyStorageClient if the Actor is running on the Apify platform,
+            # otherwise use the MemoryStorageClient.
+            storage_client = (
+                ApifyStorageClient.from_config(config) if config.is_at_home else MemoryStorageClient.from_config(config)
```

Comment:
This is supposed to happen in `Actor.init`, right? Why duplicate it here?

Reply:
Because of nested event loop, otherwise, it will result in:
```
RuntimeError: <asyncio.locks.Event object at 0x7c2d640c8fc0 [unset]> is bound to a different event loop
```
when using Apify client.